Sources/Support/PurchasesDiagnostics.swift

```diff
@@ -187,17 +196,39 @@ extension PurchasesDiagnostics {
     /// Status of the SDK Health report
     public enum SDKHealthStatus: Sendable {
         /// SDK configuration is valid but might have some non-blocking issues
-        case healthy(warnings: [PurchasesDiagnostics.Error])
+        case healthy(warnings: [PurchasesDiagnostics.SDKHealthError])
         /// SDK configuration is not valid and has issues that must be resolved
-        case unhealthy(PurchasesDiagnostics.Error)
+        case unhealthy(PurchasesDiagnostics.SDKHealthError)
+    }
+
+    /// Perform tests to ensure SDK is configured correctly.
+    /// - `Throws`: ``PurchasesDiagnostics/Error`` if any step fails
+    @available(*, deprecated, message: """
+    Use the `PurchasesDiagnostics.shared.checkSDKHealth()` method instead.
+    """)
+    @objc(testSDKHealthWithCompletion:)
+    public func testSDKHealth() async throws {
+        do {
+            try await self.unauthenticatedRequest()
+            #if !ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION
+            try await self.authenticatedRequest()
+            #endif
+            try await self.offeringsRequest()
+            try await self.signatureVerification()
+        } catch let error as Error {
+            throw error
+        } catch let error {
+            // Catch every other error to ensure that we only throw `Error`s from here.
+            throw Error.unknown(error)
+        }
     }
 
     #if DEBUG
     /// Performs a full SDK configuration health check and throws an error if the configuration is not valid.
     /// - Important: This method is intended solely for debugging configuration issues with the SDK implementation.
     /// It should not be invoked in production builds.
     /// - Throws: The specific configuration issue that needs to be solved.
```

Comment:
I would keep this method documentation in the existing (now deprecated) `testSDKHealth` method. Mostly because that method can still be called outside of `DEBUG` builds. 
And I would simplify the documentation of the new `checkSDKHealth()` method. E.g., "It should not be invoked in production builds." could be removed as now supposedly "it cannot be invoked in production builds". WDYT?

Reply:
I decided to just revert to what the documentation looked like before my changes, but I 100% agree, it's an improvement if we add this. Let me change this 👍

Reply:
Updated!